The gleep/update UI rotation speed

Earl-Brown opened the pull request:
right-clicking in the preview area now brings up an options dialog with orbit speed sensitivity and a checkbox for whether or not to display the grid.

Changes made:
* Added UIOptions type
* Added UIOptionsPopup view
* Moved ValueCheckbox and ValueSlider out to Controls folder parallel to App to make them available to UIOptionsPopup
* Renamed "Controls" to "OrbitView" to make it's purpose more clear
* Added sliderColumns and spinnerColumns properties to ValueSlider (it was not using all columns in the Options dialog)
*

accassid committed:
1 - Readme update

* Updating readme with getting started notes

accassid committed:
Update 0.1

* Initial version of DiceGen for production testing

accassid committed:
Hotfix - remove bad dependency

* Removing bad dependency from yaml

accassid committed:
Hotfix - HTTTPS Fonts

accassid committed:
Hotfix - Fix cloudbuild

accassid committed:
Update 0.1.1

21 - Orientation Indicators

accassid committed:
Update 0.1.2

32 - Dice Ratio

accassid committed:
Update 0.1.3

* External links
* Code docs
* Dependency update

accassid committed:
Update 0.2 

* 62 - Upload fonts 
* 64 - SVG position

accassid committed:
0.2.1

* Adding spindowns

accassid committed:
Update 0.2.2

accassid committed:
Udpate 0.2.3

* Coin geometry

Earl-Brown committed:
Added state property for orbit speed

Earl-Brown committed:
Things are working, just finishing up styling

Earl-Brown committed:
added show/hide grid option; styling is almost done

Earl-Brown committed:
A little more style tweaking; renamed 'Controls' to 'OrbitView'

Earl-Brown committed:
ready for prime-time!

Earl-Brown commented:
I am *sure* there will be suggestions for formatting and code style - I welcome them, and will keep an eye out to make the changes as soon as possible!

accassid commented on src/views/UIOptionsPopup/UIOptionsPopup.tsx:
```diff
@@ -0,0 +1,49 @@
+  const hidePopup = () => setVisibility(false)
+
+  const optionsDialog = isVisible ? (
+    <UIOptionsDialog>
```
I like the tenacity to create a custom dialog popup. However for a project like this where we already have UI library installed, it's generally better to just use a component already available to you rather than making one from scratch. It both reduces code overhead and adds to UI uniformity on the app. I'd recommend using the antd modal for this one. Check out the docs here [https://ant.design/components/modal/](https://ant.design/components/modal/)

Earl-Brown replied:
TBH - I was having trouble getting the modal to work ... but I've learned a bit, so it'll probably go easier when I adapt to all of your suggestions :)

Earl-Brown replied:
I have submitted changes to honor your suggestions!

accassid commented on src/views/UIOptionsPopup/UIOptionsPopup.tsx:
```diff
@@ -0,0 +1,49 @@
+  )
+
+  return (
+    <div onContextMenu={showPopup} style={{ height: 'inherit', width: 'inherit}' }}>
```
Interesting thought putting it on a context menu trigger, however I don't think that's necessary for this. First off, people wouldn't intuitively know to do that (I had to look at the code to figure it out) and second people often get cranky when they have their regular context menus overridden in the browser. To that end, I would just use a settings icon from [antd](https://ant.design/components/icon/) and make a button out of it that triggers the menu. I'd put that button either in the toolbar by the menu button, in the upper right of the settings panel, or maybe just make it another button with the social media. Play around and see what looks best.

Earl-Brown replied:
I added a configuration button, and removed the context menu trigger.

accassid commented on src/models/UIOptions.ts:
```diff
@@ -0,0 +1,11 @@
+export type UIOptions = {
+  orbitSpeed: number
+  showGrid: number
+  isUIOptionsPopupVisible: true | false
```
I'd probably just use the `boolean` type here.

Earl-Brown replied:
I have submitted changes to honor your suggestions!

accassid commented on src/views/PageWrapper/PageWrapper.tsx:
```diff
@@ -26,7 +28,7 @@ const PageWrapper: React.FC<Props> = ({ children }: Props) => {
       <PageBody>
         <RightPanel />
-        {children}
+        <UIOptionsPopup>{children}</UIOptionsPopup>
```
If you use a button triggered modal, as suggested later, no need to pass the page body children into this component for the context menu trigger.

Earl-Brown replied:
I have submitted changes to honor your suggestions!

accassid reviewed (changes requested):
Overall, good first PR. Your types, models, and file placements all seem reasonable to me. The big thing I'd like to see changed is the styling of the menu itself and its trigger as I commented. Nice work.

Earl-Brown commented:
Thanks for the feedback - I will work to make these changes.

Earl-Brown committed:
Changed menu icon to show 'collapse/expand' icons instead of just the menu

Earl-Brown committed:
Added configuration button

Earl-Brown committed:
Changed UIOptions to ConfigurationSettings

Earl-Brown committed:
Settings dialog is now connected to the configuration button

Earl-Brown commented:
I have revised the interface to honor suggestions:
* changed menu to icon so it indicates what will happen when it is clicked (collapsed or expanded)
* Added a "config" icon
* bound the configuration dialog to the config button and changed where it appears

Earl-Brown commented:
In my earlier comments, I forgot to mention that I'm leaving my original popup instead of using a modal, because I couldn't see a way to have a modal allow the user to interact with the preview; I like it better to have the users able to test the speed they've set before they close the window.

accassid commented on src/views/UIOptionsPopup/ConfigurationDialog.tsx:
```diff
@@ -0,0 +1,44 @@
+      <SettingOutlined onClick={togglePopup} style={style}></SettingOutlined>
+      {isVisible ? (
+        <UIOptionsDialog>
+          <h3 style={{ borderBottom: '1px solid white' }}>Settings</h3>
```
I would change this line color to match the border color of your UIOptionsDialog.

accassid replied:
See my color suggestion for the border in the style file.

Earl-Brown replied:
Will do

Earl-Brown replied:
Made the color changes and fixed the compile warnings.

accassid commented on src/views/UIOptionsPopup/style.ts:
```diff
@@ -0,0 +1,13 @@
+  right: 16px;
+  width: 38em;
+  position: absolute;
+  background-color: rgba(0, 0, 0, 0.9);
```
To contribute to the app's uniformity, I'd change this background and opacity to match that of the RightPanel's. I believe that should be `rgba(28, 28, 28, 0.9)`.

Earl-Brown replied:
Thanks for the suggestion!  I wasn't sure what to use.

Earl-Brown replied:
Done

accassid commented on src/views/UIOptionsPopup/style.ts:
```diff
@@ -0,0 +1,13 @@
+  width: 38em;
+  position: absolute;
+  background-color: rgba(0, 0, 0, 0.9);
+  background-opacity: 0.5;
```
Shouldn't need this property if you're using rgba.

Earl-Brown replied:
Again, thanks!

Earl-Brown replied:
Done

accassid commented on src/views/UIOptionsPopup/style.ts:
```diff
@@ -0,0 +1,13 @@
+  position: absolute;
+  background-color: rgba(0, 0, 0, 0.9);
+  background-opacity: 0.5;
+  border: 2px solid silver;
```
More uniformity points, I'd use the same grey that the UI library is using, so change `silver` to `#434343`

Earl-Brown replied:
>thumbup<

Earl-Brown replied:
Done

accassid commented on src/views/TopToolbar/TopToolbar.tsx:
```diff
@@ -36,7 +39,12 @@ const TopToolbar: React.FC<Props> = () => {
       <BarSegment>
-        <MenuOutlined onClick={togglePanel} style={{ fontSize: 25, cursor: 'pointer' }} />
+        <ConfigurationDialog style={{ ...iconStyle, margin: '0 0.5em' }} />
+        {rightPanelActive ? (
```
I like the concept of what you're doing here, however I'd remove it from this PR for two reasons:
1. It's out of the scope of this PR, so if you think it should be added open a different PR. PRs can be small!
2. This doesn't work with the rightPanelActiveMobile property which is the inverse of the rightPanelActive property, meaning the way you have this coded the icon will be backwards on mobile.

Earl-Brown replied:
OK - I'll pull it for now.

Earl-Brown replied:
Done

accassid commented on src/views/UIOptionsPopup/style.ts:
```diff
@@ -0,0 +1,13 @@
+import styled from 'styled-components'
+
+export const UIOptionsDialog = styled.div`
```
I'd add a box shadow to contribute to its floating sense:
`box-shadow: 5px 5px 8px 0px rgba(0,0,0,0.68);`

accassid replied:
Also, make sure to always check UI elements in responsive mode to make sure they work on mobile. This element looks like this right now:
![Screenshot_2021-02-24_12-12-40](https://user-images.githubusercontent.com/14295292/109053149-a37ec480-7699-11eb-8dba-44b5f67dd651.png)
You should be able to solve this by adding a `max-width: 95%` property. However, you'll have to correct some issues with the ValueSlider overflowing then.

Earl-Brown replied:
Done.  No overflow problem at all!

accassid commented on src/views/UIOptionsPopup/style.ts:
```diff
@@ -0,0 +1,13 @@
+import styled from 'styled-components'
+
+export const UIOptionsDialog = styled.div`
+  top: 40px;
```
I'd make this 43px to give it more of a floating sense.

Earl-Brown replied:
Done

accassid reviewed (changes requested):
I can see what you mean that you want to be able to test the responsiveness while the popout is open, so I'll allow it. That being said, I did request some styling changes. Also you have some warnings to resolve. Make sure before a PR that in the terminal where you ran `yarn start` there are no active warnings.
![Screenshot from 2021-02-23 14-48-39](https://user-images.githubusercontent.com/14295292/109053579-1720d180-769a-11eb-8081-9237c8290579.png)

Earl-Brown commented:
Thanks for all of the comments and suggestions!

I'll work to fix them.  Be prepared to find more issues :)

And I'll fix them as soon as I can!

Earl-Brown committed:
Merge pull request #1 from Earl-Brown/develop

* Adding beta logo and rendering it with the beta subdomain

Earl-Brown committed:
Reverted menu icon

Earl-Brown committed:
Updated styling to match pull request suggestions

Earl-Brown committed:
Fixed compile warnings

Earl-Brown committed:
Fixed rotation sensitivity spinner layout issue

Earl-Brown committed:
Merge branch 'develop' into theGleep/update-ui-rotation-speed

accassid commented:
Looks good! Seems to be just one compilation warning. Fix that up and it should be good to go. 
![Screenshot from 2021-03-08 18-08-57](https://user-images.githubusercontent.com/14295292/110402920-79ce8180-8039-11eb-9f64-acd15b155140.png)

accassid commented:
> Looks good! Seems to be just one compilation warning. Fix that up and it should be good to go.
> ![Screenshot from 2021-03-08 18-08-57](https://user-images.githubusercontent.com/14295292/110402920-79ce8180-8039-11eb-9f64-acd15b155140.png)

Actually, nevermind, that must have been on dev when you forked it and it's fixed now. Never bad to rebase before a PR though.

Earl-Brown commented:
Sweet,  Thanks! 

On March 8, 2021 6:14:56 PM MST, Andrew Cassidy <notifications@github.com> wrote:
>Merged #88 into develop.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/accassid/dice-gen/pull/88#event-4424507764

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
